### DISCUSSION TITLE
Custom crowdsourcing task

### DISCUSSION THREAD

**Discussion by Sea94** [facebookresearch/ParlAI#4025] [Q&A]
Hello!

Thanks for developing this great platform.

I would like to run some crowdsourcing experiments on Mturk and was wondering if the following use cases are supported at the moment.

- Can I have a human chat with (several) model agent(s) and another human within one task, given that conversations should happen sequentially (i.e., not as a simultaneous group chat). In other words, can I combine “Chat demo” and “Model chat” [example tasks](https://parl.ai/docs/tutorial_crowdsourcing.html) under one task?
- Can the two workers who are paired to chat with each other under one task have different instructions displayed to them/different rewards offered/different follow-up questions asked?

Thank you in advance

**Comment by mojtaba-komeili** [comment]
I recommend you using our crowdsourcing tool, [Mephisto](https://github.com/facebookresearch/Mephisto). You have full control over the agents in the world and their interactions.
Also, for your second question, see [this task](https://github.com/facebookresearch/ParlAI/tree/main/parlai/crowdsourcing/projects/wizard_of_internet) where two humans involved have different roles with different instructions and onboarding.

**Reply by Sea94** [reply]
Thanks a lot for the references, @mojtaba-komeili ! I will have a look.
If I launch a task with the help of Mephisto, is it possible to include some custom UI elements (multiple choice questions, drag'n'drop controls, etc.)?

**Reply by mojtaba-komeili** [reply]
Mephisto is is very flexible when it comes to designing customized UI for your task. I recommend you check their customized task for ParlAI chat [here](https://github.com/facebookresearch/Mephisto/tree/main/examples/parlai_chat_task_demo).
